File: data/cpp/tests/test_Pydantic_basemodel.cpp
#include <gtest/gtest.h>
#include "../src/function_Pydantic_basemodel.cpp"

class TestCreateUser : public ::testing::Test {
protected:
    UserValidator validator;
};

TEST_F(TestCreateUser, TestValidUser) {
    std::string result = validator.create_user("John Doe", 30, "john@example.com");
    EXPECT_EQ(result, "name='John Doe' age=30 email='john@example.com'");
}

TEST_F(TestCreateUser, TestInvalidAge) {
    std::string result = validator.create_user("John Doe", -1, "john@example.com");
    EXPECT_NE(result.find("age"), std::string::npos);
}

TEST_F(TestCreateUser, TestInvalidEmail) {
    std::string result = validator.create_user("John Doe", 30, "invalid-email");
    EXPECT_NE(result.find("email"), std::string::npos);
}

TEST_F(TestCreateUser, TestMissingField) {
    std::string result = validator.create_user("John Doe", 30, "");
    EXPECT_NE(result.find("email"), std::string::npos);
}

TEST_F(TestCreateUser, TestInvalidEmailType) {
    std::string result = validator.create_user("John Doe", 30, "12345");
    EXPECT_NE(result.find("email"), std::string::npos);
}

int main(int argc, char **argv) {
    ::testing::InitGoogleTest(&argc, argv);
    return RUN_ALL_TESTS();
}

File: data/python/tests/test_Pydantic_basemodel.py
import unittest
from function_Pydantic_basemodel import UserValidator

class TestCreateUser(unittest.TestCase):

    def test_valid_user(self):
        validator = UserValidator()
        result = validator.create_user("John Doe", 30, "john@example.com")
        self.assertEqual(result, "name='John Doe' age=30 email='john@example.com'")

    def test_invalid_age(self):
        validator = UserValidator()
        result = validator.create_user("John Doe", "thirty", "john@example.com")
        self.assertIn('age', result)

    def test_invalid_email(self):
        validator = UserValidator()
        result = validator.create_user("John Doe", 30, "invalid-email")
        self.assertIn('email', result)

    def test_missing_field(self):
        validator = UserValidator()
        result = validator.create_user("John Doe", 30, "")
        self.assertIn('email', result)

    def test_invalid_email_type(self):
        validator = UserValidator()
        result = validator.create_user("John Doe", 30, 12345)
        self.assertIsInstance(result, str)
        self.assertIn('email', result)

if __name__ == "__main__":
    unittest.main()
